auxx/src/Lang/Lexer.hs

```diff
@@ -141,17 +143,27 @@ detokenize = unwords . List.map tokenRender
 
 type Lexer a = Parsec Void Text a
 
-tokenize :: Text -> [Token]
+tokenize :: Text -> [(Span, Token)]
 tokenize = fromMaybe noTokenErr . tokenize'
   where
     noTokenErr =
         error "tokenize: no token could be consumed. This is a bug"
 
-tokenize' :: Text -> Maybe [Token]
+tokenize' :: Text -> Maybe [(Span, Token)]
 tokenize' = parseMaybe (between pSkip eof (many pToken))
 
-pToken :: Lexer Token
-pToken = (try pToken' <|> pUnknown) <* pSkip
+pToken :: Lexer (Span, Token)
+pToken = withPosition (try pToken' <|> pUnknown) <* pSkip
+  where
+    posToNum :: (Num a, Num b) => SourcePos -> (a, b)
```

Comment:
Why not change it to `posToLoc`? AFAIU, it's only used in combination with `loc` (`uncurry loc . posToNum`).